app/controllers/application_controller.rb

```diff
@@ -8,6 +8,6 @@ def require_user_logged_in!
   private
  
   def current_user
-    @current_user ||= User.find(session[:user_id]) if session[:user_id]
+    @current_user ||= User.find(session[:user_id]) if session[:user_id] unless User.count == 0
```

Comment:
Nested if / unless is weird structure here, and we can restructure this to get the same effect without the double conditional.

```suggestion
    @current_user ||= User.find_by(id: session[:user_id])
```